src/middlewared/middlewared/plugins/rdma/interface/crud.py

```diff
@@ -0,0 +1,188 @@
+import middlewared.sqlalchemy as sa
+from middlewared.schema import (Dict, Int, IPAddr, Patch, Str,
+                                accepts)
+from middlewared.service import CallError, CRUDService
+
+from pyroute2 import NDB
+from pyroute2.ndb.transaction import CheckProcessException
+
+
+class ConnectionChecker:
+    def __init__(self, middleware, netdev, ip, mac=None):
+        self.middleware = middleware
+        self.netdev = netdev
+        self.ip = ip
+        self.mac = mac
+
+    def commit(self):
+        if not self.middleware.call_sync('rdma.interface.local_ping', self.netdev, self.ip, self.mac):
+            raise CheckProcessException('CheckProcess failed')
+
+
+class RDMAInterfaceModel(sa.Model):
+    __tablename__ = 'rdma_interface'
+    __table_args__ = (sa.UniqueConstraint('rdmaif_node', 'rdmaif_ifname'),)
+
+    id = sa.Column(sa.Integer(), primary_key=True)
+
+    rdmaif_node = sa.Column(sa.String(120), nullable=False)
+    rdmaif_ifname = sa.Column(sa.String(120), nullable=False)
+    rdmaif_address = sa.Column(sa.String(45), nullable=False)
+    rdmaif_prefixlen = sa.Column(sa.Integer(), nullable=False)
+    rdmaif_mtu = sa.Column(sa.Integer(), nullable=False)
+
+
+class RDMAInterfaceService(CRUDService):
+
+    class Config:
+        namespace = 'rdma.interface'
+        private = True
+        cli_private = True
+        datastore = 'rdma.interface'
+        datastore_prefix = "rdmaif_"
```

Comment:
You might be able to re-use the `NETWORK_INTERFACE_` role prefix here (assuming that's appropriate).

Reply:
Currently this API is not public, so I believe that means a role is not appropriate.